base/pkg/entry.jl

```diff
@@ -719,7 +720,7 @@ function test!(pkg::AbstractString,
                     --compilecache=$(Bool(Base.JLOptions().use_compilecache) ? "yes" : "no")
                     --check-bounds=yes
                     --startup-file=$(Base.JLOptions().startupfile != 2 ? "yes" : "no")
-                    $test_path
+                    --eval $code
```

Comment:
Not strictly necessary but does make the "runtests.jl" file run inside of a module which means variables defined within the startup-file can only be accessed via `Main.*`. I made this modification to make it harder to accidentally reference a variable within the tests which was defined or imported within the startup-file.